File: 백준/Unrated/10799. 쇠막대기/쇠막대기.py
s = input()
s = s.replace('()','z')
stack = list()
ans = 0

for i in s:
    if i == '(':
        stack.append(i)
        ans += 1
    elif i == ')':
        stack.pop()
    elif i == 'z':
        ans += len(stack)
        
print(ans)

File: 백준/Silver/10799. 쇠막대기/쇠막대기.java
import java.util.*;
import java.io.*;

public class Main {
    public static void main(String[] args) throws IOException {
        Deque<Character> stack = new ArrayDeque<>();
        BufferedReader br = new BufferedReader(new InputStreamReader(System.in));
        
        String input = br.readLine();
        boolean laserOn = false;
        int count = 0;
        
        for (char c : input.toCharArray()) {
            if (c == '(') {
                laserOn = true;
                stack.push('(');
            } else {
                if (laserOn) {
                    stack.pop();
                    count += stack.size();
                    laserOn = false;
                } else {
                    stack.pop();
                    count++;
                }
            }
        }
        
        System.out.println(count);
    }
}